cli/azd/pkg/project/framework_service_dotnet.go

```diff
@@ -22,6 +22,12 @@ const (
 	defaultDotNetBuildConfiguration string = "Release"
 )
 
+// DotNet Framework Options
+type DotNetFrameworkOptions struct {
+	// When set will enable integration with dotnet user-secrets for to support local development
+	UserSecrets bool `yaml:"userSecrets"`
```

Comment:
I wonder if we should take this opportunity to revisit the strategy of adding every output from bicep into the user secrets store.  In practice, it feels like really only a subset of the outputs would be needed. I could imagine that instead of a bool here we support an array of names of outputs and just add the matching ones.  We could even expand it such that the array could contain either strings or an object that would have an outputName and a secretName property, for cases where folks what to rename the secrets as they add them (we already have some heuristics that we run to change the names slightly, perhaps giving full control to users would be helpful)

I think - however - that we could add that in a compatible way going forward (simply expand the domain of types that are allowed to be set to this variable, which shouldn't be a breaking change), but I wanted to raise it as something we might want to do today, especially if the per invocation cost of `dotnet-secrets add` is still high.

In practice, I think all our sample applications are going to set this to `true` today, because we need to inject things like the domain of CosmosDB or a SQL Server connection string into dontet-secrets, so the ASP.NET configuration system can discover them, so I'm wondering if in practice we'll be able to reap the wins that this would offer.

Does help the case where you don't care about this integration however, and that's probably common for many .NET Apps, so happy to see this go in as is.

Reply:
@ellismg another option is to author a nuget package for .NET apps to extend ConfigurationManager and support .env files, would bypass the need for using dotnet-secrets calls.

Reply:
https://github.com/dotnet/runtime/issues/49667 is an ask to add `.env` support to .NET configuration system.

Reply:
Agree that long term it would be ideal to allow the developer the options to configure which provision outputs make it to the dotnet user-secrets but feel it is beyond the scope of this PR.